File: backend/app/api/v1/endpoints/alerts.py
from typing import Optional
from datetime import datetime
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from app.core.database import get_db
from app.core.security import get_current_user
from app.models import User, Alert
from app.schemas import AlertResponse, AlertResolve

router = APIRouter()


@router.get("/")
async def get_alerts(
    page: int = Query(1, ge=1),
    per_page: int = Query(20, ge=1, le=100),
    status: Optional[str] = Query(None),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    query = db.query(Alert)
    
    if status:
        query = query.filter(Alert.status == status)
    
    total = query.count()
    skip = (page - 1) * per_page
    alerts = query.order_by(Alert.created_at.desc()).offset(skip).limit(per_page).all()
    
    return {
        "items": [AlertResponse.model_validate(a) for a in alerts],
        "total": total,
        "page": page,
        "per_page": per_page
    }


@router.put("/{alert_id}/acknowledge", response_model=AlertResponse)
async def acknowledge_alert(
    alert_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    alert = db.query(Alert).filter(Alert.id == alert_id).first()
    if alert is None:
        raise HTTPException(status_code=404, detail="ไม่พบการแจ้งเตือน")
    
    alert.status = "acknowledged"
    alert.acknowledged_by = current_user.id
    alert.acknowledged_at = datetime.utcnow()
    
    db.commit()
    db.refresh(alert)
    return alert


@router.put("/{alert_id}/resolve", response_model=AlertResponse)
async def resolve_alert(
    alert_id: int,
    resolve_data: AlertResolve,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    alert = db.query(Alert).filter(Alert.id == alert_id).first()
    if alert is None:
        raise HTTPException(status_code=404, detail="ไม่พบการแจ้งเตือน")
    
    alert.status = "resolved"
    alert.resolved_by = current_user.id
    alert.resolved_at = datetime.utcnow()
    alert.resolution_note = resolve_data.resolution_note
    
    db.commit()
    db.refresh(alert)
    return alert

File: frontend/src/services/alerts.ts
import api from './api'
import type { Alert } from '../types'

export interface AlertsListResponse {
  items: Alert[]
  total: number
  page: number
  per_page: number
}

export const alertsService = {
  async list(page = 1, perPage = 20, status?: string): Promise<AlertsListResponse> {
    const params = new URLSearchParams()
    params.set('page', String(page))
    params.set('per_page', String(perPage))
    if (status) params.set('status', status)
    const response = await api.get(`/alerts/?${params.toString()}`)
    return response.data
  },

  async acknowledge(id: number): Promise<Alert> {
    const response = await api.put(`/alerts/${id}/acknowledge`)
    return response.data
  },

  async resolve(id: number, resolution_note?: string): Promise<Alert> {
    const response = await api.put(`/alerts/${id}/resolve`, { resolution_note })
    return response.data
  },
}

